src/components/pages/MainPage.js

```diff
@@ -1,11 +1,12 @@
-import React, {useEffect, useState} from "react";
+import React, {useEffect, useState, useRef} from "react";
 import {Auth} from "aws-amplify"
 import {Button} from "react-mdl";
 import {Pages} from "../NavigationComponent";
 import {Hub} from "aws-amplify";
 
 const MainPage = ({nav}) => {
     const [isLoading, setIsLoading] = useState(true);
+    const disconnectRef = useRef(false);
```

Comment:
Why use a ref instead of another state variable? Refs are not part of the normal React dataflow and their use should be avoided if possible.

Reply:
Also, the actual issue of multiple calls to disconnect might be cause by line 43 with `Hub.listen('auth', authHandler)` being called on every component re-render, effectively registering it multiple times. Maybe that call should be moved into an effect that returns a cleanup function that removes the listener. There is an example of this in App.js.

Reply:
Because state changes are asynchronous, whereas ref changes are instant. This is what essentially locks out the other calls to disconnect, since I couldn't figure out why the disconnect method was getting called multiple times.